datafusion/physical-expr/src/physical_expr.rs

```diff
@@ -51,23 +47,15 @@ pub trait PhysicalExpr: Send + Sync + Display + Debug {
         batch: &RecordBatch,
         selection: &BooleanArray,
     ) -> Result<ColumnarValue> {
-        if selection.iter().all(|b| b == Some(true)) {
+        let filter_count = selection
```

Comment:
The upstream already performs this optimisation, see [here](https://github.com/apache/arrow-rs/blob/master/arrow/src/compute/kernels/filter.rs#L403) and so you can probably elide counting the bits again.

Reply:
I thought this as well, but the earlier fast path also avoided the `scatter` path in case of all values being true.
The filter count can not easily be reused between this code and the `filter` kernel.

I am also OK with removing this path, for making the code a bit simpler.

Reply:
> also avoided the scatter path in case of all values being true.

You could inspect the length of the returned filtered array instead of counting the bits twice?

Reply:
That is a good suggestion :+1:

Reply:
Done